python/pyspark/sql/connect/plan.py

```diff
@@ -1140,6 +1140,50 @@ def plan(self, session: "SparkConnectClient") -> proto.Relation:
         return plan
 
 
+class StatSampleBy(LogicalPlan):
+    def __init__(
+        self,
+        child: Optional["LogicalPlan"],
+        col: "ColumnOrName",
+        fractions: Dict[Any, float],
+        seed: Optional[int],
+    ) -> None:
+        super().__init__(child)
+
+        assert col is not None and isinstance(col, (Column, str))
+
+        assert fractions is not None and isinstance(fractions, dict)
+        for k, v in fractions.items():
+            assert v is not None and isinstance(v, float)
+
+        assert seed is None or isinstance(seed, int)
```

Comment:
Do we need add these check here?

Reply:
I prefer adding assertion in the plan layer to make sure all parameters are expected

Reply:
I think we can let server side require that the client must pass seed.